spec/generators/blueprint_generator_spec.rb

```diff
@@ -0,0 +1,172 @@
+      end
+    end
+
+    describe "given -d" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} -d wherevs)
+      end
+
+      subject { file('wherevs/vehicle_blueprint.rb') }
+
+      it "file is created where we say" do
+        is_expected.to exist
+      end
+    end
+
+    describe "given -i" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} -i)
+      end
+
+      it "blueprint file has identifier" do
+        is_expected.to contain(/identifier/)
+      end
+    end
+
+    describe "given --fields=yoohoo hi_there" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} --fields=yoohoo hi_there)
+      end
+
+      it "blueprint file has manual fields" do
+        is_expected.to contain(/fields :yoohoo, :hi_there/)
+      end
+    end
+
+    describe "given --detect_fields" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} --detect_fields)
+      end
+
+      it "blueprint file has detected fields" do
+        is_expected.to contain(/:make, :model, :miles/)
+      end
+    end
+
+    describe "given -a=not_really" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} -a=not_really)
+      end
+
+      it "blueprint file has manual association" do
+        is_expected.to contain(/association :not_really/)
+        is_expected.to contain(/blueprint: NotReallyBlueprint/)
+      end
+    end
+
+    describe "given -a=user" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} -a=user)
+      end
+
+      it "blueprint file has manual association" do
+        is_expected.to contain(/association :user/)
+        is_expected.to contain(/blueprint: UserBlueprint/)
+      end
+    end
+
+    describe "given --detect_associations" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} --detect_associations)
+      end
+
+      it "blueprint file has detected association" do
+        is_expected.to contain(/association :user/)
+        is_expected.to contain(/blueprint: UserBlueprint/)
+      end
+    end
+
+    describe "given --detect_associations --dynamic_association" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} --detect_associations --dynamic_association)
+      end
+
+      it "blueprint file has detected association" do
+        is_expected.to contain(/blueprint: ->\(user\) {user.blueprint}/)
+      end
+    end
+
+    describe "given --detect_associations --default_association" do
+      include_examples "generated_file"
+      before do
+        run_generator %W(#{model} --detect_associations --default_association)
+      end
+
+      it "blueprint file has default association" do
+        is_expected.to contain(/default: {}/)
+      end
+    end
```

Comment:
While indeed the `{}` was used as an example in the README, it is not always the case. Devs can specify any kind of defaults for example:
```
association :friends, blueprint: FriendBlueprint, default: []
association :pictures, blueprint: PictureBlueprint, default: {}
association :posts, blueprint: PostBlueprint, default: nil
association :mice, blueprint: MouseBlueprint, default: ["Minnie Mouse", "Mickey Mouse"]
```

To be honest, I think it's OK to not support association defaults via the generator.